wayback/tests/test_client.py

```diff
@@ -623,3 +624,75 @@ def test_timeout_empty(self, mock_class):
         # Overwriting the default
         res = session.request('GET', 'http://test.com', timeout=1)
         assert res.text == '1'
+
+    @ia_vcr.use_cassette()
+    def test_search_rate_limits(self):
+        # The timing relies on the cassettes being present,
+        # therefore the first run might fail.
+        # Since another test might run before this one,
+        # I have to wait one second before starting.
+        time.sleep(1)
```

Comment:
Since you aren’t using the fancier locking class I mentioned in the issue (https://github.com/edgi-govdata-archiving/wayback/issues/12#issuecomment-787589273), I’m thinking a good solution here is to add an instance-specific prefix to the `group` name on the `_utils.rate_limited()` calls, e.g:

```py
with _utils.rate_limited(calls_per_second=calls_per_second if calls_per_second is not None else
                                          self.session.memento_calls_per_second,
                         group=f'{id(self.session)}:get_memento'):
    # rate-limited code here
```

Reply:
Hmmm, actually, I’m not sure I should do this without having a thread-safe client/session. The current approach forces all sessions to share the same timings across threads (that is, they all see the same “last time called” when determining how long to wait), which is actually a good thing if you’ve got multiple threads working in parallel, each with one client. We don’t want that situation to accidentally cause you to exceed Wayback’s rate limits.

On the flip side, this makes it harder to have independent clients following independent limits if that’s actually what you want… but I think that’s probably rare. I need to noodle on this a bit more, but I’m guessing the fancier locking class mentioned above might work better for doing both cases, but it’s probably better to tackle that as a separate PR.